src/hotspot/share/opto/regmask.hpp

```diff
@@ -213,16 +492,25 @@ class RegMask {
   // Verify watermarks are sane, i.e., within bounds and that no
   // register words below or above the watermarks have bits set.
   bool valid_watermarks() const {
-    assert(_hwm < _RM_SIZE, "_hwm out of range: %d", _hwm);
-    assert(_lwm < _RM_SIZE, "_lwm out of range: %d", _lwm);
+    assert(_hwm < _rm_size, "_hwm out of range: %d", _hwm);
+    assert(_lwm < _rm_size, "_lwm out of range: %d", _lwm);
     for (unsigned i = 0; i < _lwm; i++) {
-      assert(_RM_UP[i] == 0, "_lwm too high: %d regs at: %d", _lwm, i);
+      assert(_rm_up(i) == 0, "_lwm too high: %d regs at: %d", _lwm, i);
     }
-    for (unsigned i = _hwm + 1; i < _RM_SIZE; i++) {
-      assert(_RM_UP[i] == 0, "_hwm too low: %d regs at: %d", _hwm, i);
+    for (unsigned i = _hwm + 1; i < _rm_size; i++) {
+      assert(_rm_up(i) == 0, "_hwm too low: %d regs at: %d", _hwm, i);
     }
     return true;
   }
+
+  bool is_AllStack_only() const {
+    assert(valid_watermarks(), "sanity");
+    uintptr_t tmp = 0;
+    for (unsigned int i = _lwm; i <= _hwm; i++) {
+      tmp |= _rm_up(i);
+    }
+    return !tmp && is_AllStack();
```

Comment:
Implicit null checks?

Could you not check `is_AllStack` early and return already?

Reply:
Thanks, I added an early exit for `is_AllStack` and also an early exit in the loop.